test/test_scripttofilename.py

```diff
@@ -93,21 +102,21 @@ def _test_ascii_windows_compatibility(self):
         metadata['artist'] = '\u2216/\\\u2215'
         settings = config.setting.copy()
         settings['ascii_filenames'] = True
-        filename = script_to_filename('%artist%/\u2216\\\\\u2215', metadata, settings=settings)
+        (filename, new_metadata) = script_to_filename('%artist%/\u2216\\\\\u2215', metadata, settings=settings)
         self.assertEqual('____/_\\_', filename)
 
     def test_remove_null_chars(self):
         metadata = Metadata()
-        filename = script_to_filename('a\x00b\x00', metadata)
+        (filename, new_metadata) = script_to_filename('a\x00b\x00', metadata)
         self.assertEqual('ab', filename)
 
     def test_remove_tabs_and_linebreaks_chars(self):
         metadata = Metadata()
-        filename = script_to_filename('a\tb\nc', metadata)
+        (filename, new_metadata) = script_to_filename('a\tb\nc', metadata)
         self.assertEqual('abc', filename)
 
     def test_preserve_leading_and_trailing_whitespace(self):
         metadata = Metadata()
         metadata['artist'] = 'The Artist'
-        filename = script_to_filename(' %artist% ', metadata)
+        (filename, new_metadata) = script_to_filename(' %artist% ', metadata)
```

Comment:
I wonder if we should the following:
rename `script_to_filename()` to `script_to_filename_with_metadata()`
provide a new wrapper function named `script_to_filename()` calling `script_to_filename_with_metadata()` but ignoring the second returned value:

```python
def script_to_filename(...):
    (filename, _unused) = script_to_filename_with_metadata(...)
    return filename
```

Reply:
I like this very much, changed the code. I also added docstrings to the functions.